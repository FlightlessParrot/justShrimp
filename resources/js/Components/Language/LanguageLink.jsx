import { Link } from "@inertiajs/react";

export default function LanguageLink({text,bgUrl, url}) {

  return (
    <Link className={`relative group hover:scale-110 rounded-md flex justify-center items-center    transition-all`} href={url} aria-label={'zmień język na / change language '+text} style={{
      backgroundImage: `url(${bgUrl})`,
      width: '80px',
      height: '40px'
    }}><span className="relative z-20 invisible group-hover:visible transition-all text-white"> {text}</span>  
      <div className="absolute w-full h-fit  group-hover:bg-none opacity-50 group-hover:bg-black group-hover:h-full transition-all">
     
     </div>
    </Link>
  )
}