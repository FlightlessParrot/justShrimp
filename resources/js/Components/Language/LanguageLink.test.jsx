import { render } from "@testing-library/react";

import { describe, expect } from "vitest";
import LanguageLink from "./LanguageLink";



describe('test Language Button component',()=>{
test("component renders ", () => {
    const snap = render(<LanguageLink text={"text"}  />);

    expect(snap).toMatchSnapshot();
});

})