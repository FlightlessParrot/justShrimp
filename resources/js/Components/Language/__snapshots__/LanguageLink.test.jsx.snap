// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Language Button component > component renders  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="zmień język na / change language text"
        class="relative group hover:scale-110 rounded-md flex justify-center items-center    transition-all"
        href=""
        style="background-image: url(undefined); width: 80px; height: 40px;"
      >
        <span
          class="relative z-20 invisible group-hover:visible transition-all text-white"
        >
           
          text
        </span>
        <div
          class="absolute w-full h-fit  group-hover:bg-none opacity-50 group-hover:bg-black group-hover:h-full transition-all"
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="zmień język na / change language text"
      class="relative group hover:scale-110 rounded-md flex justify-center items-center    transition-all"
      href=""
      style="background-image: url(undefined); width: 80px; height: 40px;"
    >
      <span
        class="relative z-20 invisible group-hover:visible transition-all text-white"
      >
         
        text
      </span>
      <div
        class="absolute w-full h-fit  group-hover:bg-none opacity-50 group-hover:bg-black group-hover:h-full transition-all"
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
